### ISSUE TITLE
Changing an NLB service from externalTrafficPolicy Cluster to Local results in a stuck Service

### ISSUE THREAD

**Issue by rbtcollins** [kubernetes/kubernetes#80996]
**What happened**:

I had an NLB Service object (in AWS naturally) where we decided we needed the source IP address and so changed the ExternalTrafficPolicy from Cluster to Local, but didn't delete the Service object - and didn't want to, because that would break Route53 DNS entries.

The Service started throwing Warning and 400 errors from the AWS API and did not self-heal.
```
  Warning  CreatingLoadBalancerFailed  9m17s               service-controller  Error creating load balancer (will retry): failed to ensure load balancer for service XXX/YYY: Error modifying target group health check: "InvalidConfigurationRequest: You cannot change the health check protocol for a target group with the TCP protocol\n\tstatus code: 400, request id: f8e9446d-b7d3-11e9-886c-3f71b1696d69"
```

**What you expected to happen**:
The controller should delete out the invalid target group and recreate it in order to configure the NLB as requested.

**How to reproduce it (as minimally and precisely as possible)**:
* Create a LoadBalancer Service annotated as an NLB with default ExternalTrafficPolicy, backed by some running pods.
* Wait for the NLB to be fully provisioned.
* Change the service to have ExternalTrafficPolicy Local.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.13 (eks latest version)
- Cloud provider or hardware configuration: AWS
- OS (e.g: `cat /etc/os-release`): unknown - eks control plane
- Kernel (e.g. `uname -a`): unknown - eks control plane
- Install tools: eksctl
- Network plugin and version (if this is a network-related bug): unknown - eks control plane
- Others:

**Comment by rbtcollins** [comment]
@kubernetes/sig-aws-bugs

**Comment by rbtcollins** [comment]
I worked around this by manually deleting the listener and then the target group; this stopped traffic until the controller retried, but saved DNS shenanigans

**Comment by felixbuenemann** [comment]
I got the same error when upgrading from Kubernetes 1.15.6 to 1.16.4 where the load balancer was already using `externalTrafficPolicy: Local`, so just upgrading broke the load balancer.

I had to delete the NLB and then switch between Cluster and Local to get it to re-register the listeners.

**Comment by kireledan** [comment]
I was able to replicate this in the master branch. I'll investigate a fix for this.

**Comment by SaranBalaji90** [comment]
When externalTrafficPolicy is local, HTTP protocol is used for health check and when its cluster  mode, TCP protocol is used. Looks like NLB doesn't support updating the protocol. So instead of updating the TG we should recreate the TG with right protocol.

Code reference 
For Local policy
https://github.com/kubernetes/kubernetes/blob/fbcd0f84d83ca6099c19176a82f20ae67d4a4f97/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L3716-L3720

**Comment by TBBle** [comment]
Per discussion on https://github.com/kubernetes/cloud-provider-aws/issues/87, this is still a problem in 1.14 and 1.15, and the [relevant code](https://github.com/kubernetes/kubernetes/blob/b67e7d6e2cacb1af3b4f366531166dfe71c2866a/staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go#L530) hasn't changed in master.

I posted more details in https://github.com/kubernetes/cloud-provider-aws/issues/87#issuecomment-648603010, but in-short we need a simliar fix for a few of the TargetGroup config options as was done in https://github.com/kubernetes/kubernetes/pull/89562 for the nodePort.

**Comment by TBBle** [comment]
/reopen

Still waiting on #92556 which adds the health-check protocol and interval to the target group name in `buildTargetGroup`.

**Comment by TBBle** [comment]
Fix in #94546 is part of Kubernetes 1.20.
